src/engine.cpp

```diff
@@ -257,6 +257,17 @@ const char *Engine_GetBasePath()
 }
 
 
+const char *Engine_GetSDLError(const char* message)
+{
+    const char *sdlMessage = SDL_GetError();
+    char *errorMessage = new char[strlen(message) + strlen(sdlMessage)];
+    strcpy(errorMessage, message);
+    strcat(errorMessage, sdlMessage);
```

Comment:
This will write the NUL character out-of-bounds, because the buffer is 1 character too short.

Apart from that, I agree with @TeslaRus.

Reply:
nice!

Reply:
It printed correctly even if you're right that it's 1 char short. Done it without reinventing the wheel though.